### PR TITLE
tests: split the nested helper in 2 parts

### PR DESCRIPTION
**Pull request by sergiocazzolato**
The idea of this change is to split the nested helper which currently y
really big in 2 parts.

The first part is the nested-tool.sh which in a following pr will be
converted to a tool. This tools contains the preparation/restoration of
the environment and the vms, and some functions needed to support those.

The second part is the nested,sh which contains the common functions
used by the tests.

This is the first step of the migration to the nested tool.

### CODE REVIEW

File: spread.yaml [hunk 1]
```diff
@@ -1045,15 +1044,15 @@ suites:
...
 
-            #shellcheck source=tests/lib/nested.sh
-            . "$TESTSLIB/nested.sh"
+            #shellcheck source=tests/lib/nested-tool.sh
+            . "$TESTSLIB/nested-tool.sh"
+
             nested_prepare_env
             nested_create_core_vm
             nested_start_core_vm
-            nested_wait_for_snap_command
```
[the remark this answers is not in the document]
Comment (sergiocazzolato) [reply spread.yaml:1053]: I moved that to the nested_start_core_vm function, as the last step to make sure the vm is ready

File: tests/nested/manual/preseed/task.yaml [hunk 2]
```diff
@@ -11,111 +11,116 @@ environment:
...
+
+    # for images that are already preseeded, we need to undo the preseeding there
+    echo "Running preseed --reset for already preseeded cloud images"
+    SNAPD_DEBUG=1 /usr/lib/snapd/snap-preseed --reset "$IMAGE_MOUNTPOINT"
 
 restore: |
-  #shellcheck source=tests/lib/nested.sh
-  . "$TESTSLIB/nested.sh"
-  nested_destroy_vm
+    #shellcheck source=tests/lib/nested-tool.sh
```
Comment (zyga) [inline tests/nested/manual/preseed/task.yaml:52]: With both changes to nested and changes to indentation this is pretty hard to read.

**Comment by zyga** [comment]
I don't quite get what this is trying to achieve. I was suggesting to create a new tool, tests.nested, that can source nested.sh and expose a CLI interface. Then tests can adapt to use that tool, even one test at a time, and eventually the shared .sh file can be removed.

**Comment by sergiocazzolato** [comment]
@zyga thanks for the review, the idea of this change is to reduce the size of the nested helper first by spliting it, then I'l create the tool. 
I prefer to split it first and then create the tool to avoid have duplicated scripts to maintain just 1 because nested.sh is changing constantly.
Once I have the split done, I'll follow the process that you mention but just with  the nested-tool.sh

File: tests/nested/manual/preseed/task.yaml (hunk 2 above)
Comment (stolowski) [reply tests/nested/manual/preseed/task.yaml:52]: +1, this seems to contain changes that are not strictly related and change indentation but also re-shuffle the order of operations. Could you please propose this separately?

**Review by stolowski — COMMENTED** [review]
I share @zyga 's view from his earlier comment, I'm unclear how does the split help and why not simply source nested.sh into a new tool and start from there. Since nested-tool.sh just moves some functions out from nested.sh and still depends on it, I guess you will still run into issues where the tool needs to modify things that are shared, and therefore you may need to temporarily introduce new implementations of some of the helper functions (not to break tests that use these helpers directly), and those could very well live in nested.sh.

Also, I think you could avoid touching the tests at all right now until the tool becomes available, even if the split is deemed neccessary, the tests could still include nested.sh, but it would just include two new .sh files (e.g. nested-hepers.sh and nested-tool.sh). They could all be renamed at the end of the process.

**Comment by sergiocazzolato** [comment]
This pr is close because a new approach has been implemented

**Pull request closed without merging**